components/Mousepad.tsx

```diff
@@ -18,101 +20,117 @@ const browser_to_pictrl_clicks = new Map<
 };
 
-const MOUSE_DELAY_FOR_CLICK = 200; // ms
+const MOUSE_DELAY_FOR_CLICK = 100; // ms
+const MOUSE_DOWN_PACKET = getMouseClickEventCommand(
+  browser_to_pictrl_clicks.get(BROWSER_MOUSE_BUTTON.MAIN)!,
+  PICTRL_MOUSE_CLICK.DOWN,
+);
+const MOUSE_UP_PACKET = getMouseClickEventCommand(
+  browser_to_pictrl_clicks.get(BROWSER_MOUSE_BUTTON.MAIN)!,
+  PICTRL_MOUSE_CLICK.UP,
+);
 
 export function MousePad({ conn }: PiConnectionProps) {
   let prev: PreviousPointerEvent = {
-    id: -1,
-    lastPos: {
-      x: -1,
-      y: -1,
-    },
-    lastMouseDown: 0
+    panId: "",
+    lastMouseDown: 0,
+    lastMouseUp: 0,
+    doubleTapped: false
   };
 
   let delta: RelMouseMove = {
     x: 0,
     y: 0,
   };
-  return (
-    <View
-      style={{
-        flexDirection: "column",
-        flex: 1,
-        padding: 10,
-        paddingTop: 30,
-      }}
-    >
-      <View
-        onContextMenu={(e: PointerEvent) => {
-          e.preventDefault();
-        }}
-        onPointerDown={(e: PointerEvent) => {
-          prev.id = e.pointerId;
-          prev.lastPos.x = e.screenX;
-          prev.lastPos.y = e.screenY;
-          let button = e.button as BROWSER_MOUSE_BUTTON;
-          if (!browser_to_pictrl_clicks.has(button)) {
-            return;
-          }
-          prev.lastMouseDown = Date.now();
-        }}
-        onPointerUp={(e: PointerEvent) => {
-          if (e.pointerId !== prev.id) {
-            return;
-          }
+  const panResponder = React.useRef(
+    PanResponder.create({
+      // Ask to be the responder:
+      onStartShouldSetPanResponder: (evt, gestureState) => true,
+      onStartShouldSetPanResponderCapture: (evt, gestureState) =>
+        true,
+      onMoveShouldSetPanResponder: (evt, gestureState) => true,
+      onMoveShouldSetPanResponderCapture: (evt, gestureState) =>
+        true,
 
-          let button = e.button as BROWSER_MOUSE_BUTTON;
-          if (!browser_to_pictrl_clicks.has(button)) {
-            return;
-          }
+      onPanResponderGrant: (e, gestureState) => {
+        // The gesture has started. Show visual feedback so the user knows
+        // what is happening!
+        // gestureState.d{x,y} will be set to zero now
+        if (e.nativeEvent.touches.length != 1) {
+          // Work on double fingered taps etc. later
+          return;
+        }
 
-          if (Date.now() - prev.lastMouseDown < MOUSE_DELAY_FOR_CLICK) {
-            const mouseDownPacket = getMouseClickEventCommand(
-              browser_to_pictrl_clicks.get(button)!,
-              PICTRL_MOUSE_CLICK.DOWN,
-            );
-            const mouseUpPacket = getMouseClickEventCommand(
-              browser_to_pictrl_clicks.get(button)!,
-              PICTRL_MOUSE_CLICK.UP,
-            );
-            conn.send(mouseDownPacket);
-            conn.send(mouseUpPacket);
-          }
+        // User double tapped
+        if (e.nativeEvent.timestamp - prev.lastMouseUp < MOUSE_DELAY_FOR_CLICK) {
```

Comment:
Really sick and, imo, the most intuitive way to implement a "double-tap-to-click."